Instead of writing the manifest files to the directory above target_rootfs,
we write it into the install/tmp directory.

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

diff -ur oe-core.orig/meta/classes/package_rpm.bbclass oe-core/meta/classes/package_rpm.bbclass
--- oe-core.orig/meta/classes/package_rpm.bbclass	2013-01-22 17:50:10.000000000 -0600
+++ oe-core/meta/classes/package_rpm.bbclass	2013-01-22 18:08:33.403917544 -0600
@@ -422,7 +422,7 @@
 	if [ "${INC_RPM_IMAGE_GEN}" = "1" -a "${INSTALL_COMPLEMENTARY_RPM}" != "1" ]; then
 		# Dump the new solution
 		echo "Note: creating install solution for incremental install"
-		smart --data-dir=${target_rootfs}/var/lib/smart install -y --dump ${pkgs_to_install} 2> ${target_rootfs}/../solution.manifest
+		smart --data-dir=${target_rootfs}/var/lib/smart install -y --dump ${pkgs_to_install} 2> ${target_rootfs}/install/tmp/solution.manifest
 	fi
 
 	if [ "${INSTALL_COMPLEMENTARY_RPM}" != "1" ]; then
@@ -439,18 +439,18 @@
 
 		# Dump what is already installed
 		echo "Note: dump installed packages for incremental update"
-		smart --data-dir=${target_rootfs}/var/lib/smart query --installed --output ${target_rootfs}/../installed.manifest
+		smart --data-dir=${target_rootfs}/var/lib/smart query --installed --output ${target_rootfs}/install/tmp/installed.manifest
 
-		sort ${target_rootfs}/../installed.manifest > ${target_rootfs}/../installed.manifest.sorted
-		sort ${target_rootfs}/../solution.manifest > ${target_rootfs}/../solution.manifest.sorted
+		sort ${target_rootfs}/install/tmp/installed.manifest > ${target_rootfs}/install/tmp/installed.manifest.sorted
+		sort ${target_rootfs}/install/tmp/solution.manifest > ${target_rootfs}/install/tmp/solution.manifest.sorted
 		
-		comm -1 -3 ${target_rootfs}/../solution.manifest.sorted ${target_rootfs}/../installed.manifest.sorted \
-			> ${target_rootfs}/../remove.list
-		comm -2 -3 ${target_rootfs}/../solution.manifest.sorted ${target_rootfs}/../installed.manifest.sorted \
-			> ${target_rootfs}/../install.list
+		comm -1 -3 ${target_rootfs}/install/tmp/solution.manifest.sorted ${target_rootfs}/install/tmp/installed.manifest.sorted \
+			> ${target_rootfs}/install/tmp/remove.list
+		comm -2 -3 ${target_rootfs}/install/tmp/solution.manifest.sorted ${target_rootfs}/install/tmp/installed.manifest.sorted \
+			> ${target_rootfs}/install/tmp/install.list
 		
-		pkgs_to_remove=`cat ${target_rootfs}/../remove.list | xargs echo`
-		pkgs_to_install=`cat ${target_rootfs}/../install.list | xargs echo`
+		pkgs_to_remove=`cat ${target_rootfs}/install/tmp/remove.list | xargs echo`
+		pkgs_to_install=`cat ${target_rootfs}/install/tmp/install.list | xargs echo`
 		
 		echo "Note: to be removed: ${pkgs_to_remove}"
 
